--- a/net/minecraft/client/multiplayer/ClientConfigurationPacketListenerImpl.java
+++ b/net/minecraft/client/multiplayer/ClientConfigurationPacketListenerImpl.java
@@ -121,7 +_,7 @@
         );
         this.connection
             .setupInboundProtocol(
-                GameProtocols.CLIENTBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(registryaccess$frozen)),
+                GameProtocols.CLIENTBOUND_TEMPLATE.bind(net.minecraftforge.network.IForgeRegistryFriendlyByteBuf.decorator(registryaccess$frozen, this.connection)),
                 new ClientPacketListener(
                     this.minecraft,
                     this.connection,
@@ -142,7 +_,8 @@
                 )
             );
         this.connection.send(ServerboundFinishConfigurationPacket.INSTANCE);
-        this.connection.setupOutboundProtocol(GameProtocols.SERVERBOUND_TEMPLATE.bind(RegistryFriendlyByteBuf.decorator(registryaccess$frozen)));
+        this.connection.setupOutboundProtocol(GameProtocols.SERVERBOUND_TEMPLATE.bind(net.minecraftforge.network.IForgeRegistryFriendlyByteBuf.decorator(registryaccess$frozen, this.connection)));
+        net.minecraftforge.common.ForgeHooks.handleClientConfigurationComplete(this.connection);
     }
 
     @Override
